src/test/base/ERC4626Proxy.st.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {ERC4626Test} from "../std-test/ERC4626.test.sol";
+import {PoolMock} from "../aave-v3/mocks/PoolMock.sol";
+import {RewardsControllerMock} from "../aave-v3/mocks/RewardsControllerMock.sol";
+import {IPool} from "../../aave-v3/external/IPool.sol";
+import {AaveV3ERC4626} from "../../aave-v3/AaveV3ERC4626.sol";
+import {AaveV3ERC4626Factory} from "../../aave-v3/AaveV3ERC4626Factory.sol";
+import {IRewardsController} from "../../aave-v3/external/IRewardsController.sol";
+import {ERC20Mock} from "../mocks/ERC20Mock.sol";
+import {ERC4626Proxy, ERC4626} from "../../base/ERC4626Proxy.sol";
+
+contract ERC4626ProxyTest is ERC4626Test {
```

Comment:
This test was based on the tests created by a16z's team [here](https://github.com/daejunpark/yield-daddy/pull/1/files)